writer/openapi/utils.go

```diff
@@ -15,23 +15,51 @@ type TypeInfo struct {
 }
 
 func openAPIType(p *schema.Property) (TypeInfo, error) {
-	if(p.Type == schema.Type_ARRAY) {
-		at, err := openAPIType_helper(p.GetArrayPrimitiveType(), p.GetArrayObjectType())
-		if(err != nil) {
-			return TypeInfo{}, nil
-		}
-		return TypeInfo{
-			openapi_type: "array",
-			array_type: &at,
-		}, nil
+	switch p.GetTypes().(type) {
+		case *schema.Property_ArrayType:
+			return openAPITypeArray(p.GetArrayType())
+		case *schema.Property_ObjectType:
+			return openAPITypeObject(p.GetObjectType())
+		case *schema.Property_Type:
+			return openAPITypePrimitive(p.GetType())
+		default:
+			return TypeInfo{}, fmt.Errorf("reached end of openapi type switch statement")
```

Comment:
```suggestion
			return TypeInfo{}, fmt.Errorf("openapi type for %q not found", p.GetTypes().(type))
```

Or something like that - something that would better explain to the user the nature of the error they ran into.